sbi/inference/trainers/npe/npe_a.py

```diff
@@ -53,7 +54,9 @@ def __init__(
         self,
         prior: Optional[Distribution] = None,
         density_estimator: Union[
-            Literal["mdn_snpe_a"], DensityEstimatorBuilder
+            Literal["mdn_snpe_a"],
+            DensityEstimatorBuilder[ConditionalVectorFieldEstimator],
```

Comment:
another special case: npe_a can only be used with MDNs, not vector fields, not flows. therefore, we cannot support the vector field estimators here.